backend/scripts/delete-private-groups.ts

```diff
@@ -0,0 +1,93 @@
+import { convertContract } from 'common/supabase/contracts'
+import { runScript } from 'run-script'
+import { setAdjustProfitFromResolvedMarkets } from 'shared/helpers/user-contract-metrics'
+import { resolveMarketHelper } from 'shared/resolve-market-helpers'
+import { getUser } from 'shared/utils'
+
+runScript(async ({ pg }) => {
+  console.log(`deleting all private comments`)
+  await pg.none(`delete from contract_comments where visibility = 'private'`)
+
+  const sinclair = await getUser('0k1suGSJKVUnHbCPEhHNpgZPkUP2')
+  if (!sinclair) {
+    console.error('Sinclair not found')
+    process.exit(1)
+  }
+
+  // Note that all private multi markets have been resolved by now, so this should work
+  console.log('N/A all private binary markets')
+  const unresolvedContracts = await pg.map(
+    `select * from contracts where visibility = 'private'
+    and resolution is null
+    and mechanism != 'none'`,
+    [],
+    convertContract
+  )
+
+  for (const contract of unresolvedContracts) {
+    const creator = await getUser(contract.creatorId)
+    if (!creator) {
+      console.error(`Creator not found for contract ${contract.id}`)
+      continue
+    }
+    await resolveMarketHelper(contract, sinclair, creator, {
+      outcome: 'CANCELLED',
+    })
+    await setAdjustProfitFromResolvedMarkets(contract.id)
+  }
+
+  console.log(`redacting all answers`)
+  await pg.none(
+    `with private_contracts as (
+      select id from contracts where visibility = 'private' and mechanism = 'cpmm-multi-1'
+    ),
+    update answers
+    set data = data || jsonb_build_object('text', '[deleted answer ' + index + ']')
+    where contract_id in (select id from private_contracts)`
+  )
+
+  console.log(`overwriting and marking deleted all private contracts`)
+  await pg.none(
+    `update contracts
+    set data = data || '{
+        "wasPrivate": true,
+        "visibility": "unlisted",
+        "isRanked": false,
+        "deleted": true
+        "question": "Private Question (deleted)",
```

Comment:
I think we should ovewrite the slug as well

Reply:
Nvm I see you did

Reply:
it is overwritten on line 63